src/util/emitUtils.js

```diff
@@ -594,6 +596,8 @@ function doEmit(nsp, key, obj) {
           pubSubStats.trackClient(key, obj, time)
         });
       });
+  } else if (key.startsWith('refocus.internal.realtime.bot')) {
+    nsp.emit(key, newObjectAsString);
```

Comment:
I don't think we need to add a separate if for each event type.

Originally line 590 was a plain else, so when the toggle was enabled it would do the emit with callback for all events. Deepanshu added the condition in his logging changes, since it now includes the call to trackClient, which should only be
called for sample events. It probably does make sense to only do the emit with callback for sample events, since those are the only ones we're tracking that precisely, but then the toggle needs to only be checked for sample events.

So we should make the enableClientStats toggle explicitly sample-specific and only do the check for sample events:
```
 if (sampleEvent && enableSampleClientStats) {
    emit with callback
  } else {
    normal emit
  }
```

Reply:
Good catch! I think the emit with callback code was written before the filtering code, and didn't get updated to work with the filtering.